react/features/invite/components/dial-in-summary/native/DialInSummary.js

```diff
@@ -0,0 +1,156 @@
+import { getDialInfoPageURLForURIString } from '../../../functions';
+
+import DialInSummaryErrorDialog from './DialInSummaryErrorDialog';
+import styles, { INDICATOR_COLOR } from './styles';
+
+type Props = {
+
+    /**
+     * The URL to display the summary for.
+     */
+    _summaryUrl: ?string,
+
+    dispatch: Dispatch<any>
+};
+
+/**
+ * Implements a React native component that displays the dial in info page for a specific room.
+ */
+class DialInSummary extends Component<Props> {
+
+    /**
+     * Initializes a new instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onCloseView = this._onCloseView.bind(this);
+        this._onError = this._onError.bind(this);
+        this._onNavigate = this._onNavigate.bind(this);
+        this._renderLoading = this._renderLoading.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { _summaryUrl } = this.props;
+
+        return (
+            <SlidingView
+                position = 'bottom'
+                show = { Boolean(_summaryUrl) } >
+                <View style = { styles.webViewWrapper }>
+                    <HeaderWithNavigation
+                        headerLabelKey = 'info.label'
+                        onPressBack = { this._onCloseView } />
+                    <WebView
+                        onError = { this._onError }
+                        onShouldStartLoadWithRequest = { this._onNavigate }
+                        renderLoading = { this._renderLoading }
+                        source = {{ uri: getDialInfoPageURLForURIString(_summaryUrl) }}
+                        startInLoadingState = { true }
+                        style = { styles.webView } />
+                </View>
+            </SlidingView>
+        );
+    }
+
+    _onCloseView: () => void;
+
+    /**
+     * Closes the view.
+     *
+     * @returns {void}
+     */
+    _onCloseView() {
+        this.props.dispatch(hideDialInSummary());
+    }
+
+    _onError: () => void;
+
+    /**
+     * Callback to handle the error if the page fails to load.
+     *
+     * @returns {void}
+     */
+    _onError() {
+        this.props.dispatch(hideDialInSummary());
+        this.props.dispatch(openDialog(DialInSummaryErrorDialog));
+    }
+
+    _onNavigate: Object => Boolean;
+
+    /**
+     * Callback to intercept navigation inside the webview and make the native app handle the dial requests.
+     *
+     * NOTE: We don't navigate to anywhere else form that view.
+     *
+     * @param {any} request - The request object.
+     * @returns {boolean}
+     */
+    _onNavigate(request) {
+        const { url } = request;
+
+        if (url.startsWith('tel:')) {
+            Linking.openURL(url);
+            this.props.dispatch(hideDialInSummary());
+        }
+
+        return url === getDialInfoPageURLForURIString(this.props._summaryUrl);
```

Comment:
what does the return value do?

Reply:
This is to only let the webview navigate to the info page and nowhere else